.github/workflows/cairo.yml

```diff
@@ -19,6 +19,17 @@ jobs:
       - run: scarb fmt --check
         working-directory: compiler-rt/
 
+  build-check:
+    name: "Analyze the current package and report errors, but don't build Sierra files"
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: software-mansion/setup-scarb@v1
+        with:
+          scarb-version: "2.10.1"
+      - run: scarb check
+        working-directory: compiler-rt/
```

Comment:
Does `scarb check` fail if there are any warnings? We want this step to fail on errors _and_ warnings if at all possible.